Source/Public/Clear-BicepModuleCache.ps1

```diff
@@ -0,0 +1,115 @@
+            TestModuleVersion
+        }       
+    }
+
+    process {
+        
+        if ($Oci) {            
+            $OciPath = Get-BicepNetCachePath -Oci 
+            $RepositoryPath = $Repository -replace '\\', '$'
+
+            if (($Registry) -and ($Repository) -and ($Version)) {
+                Remove-Item -Recurse -Path "$OciPath/$Registry/$RepositoryPath/$Version$"
+                Write-Verbose "Cleared version [$Version] of [$Repository] in [$Registry] from local module cache"
+            }
+            elseif (($Registry) -and ($Repository)) {
+                Remove-Item -Recurse -Path "$OciPath/$Registry/$RepositoryPath"
+                Write-Verbose "Cleared [$Repository] in [$Registry] from local module cache"
+            }
+            elseif ($Registry) {
+                Remove-Item -Recurse -Path "$OciPath/$Registry"
+                Write-Verbose "Cleared [$Registry] from local module cache" 
+            }
+            else {
+                if (Test-Path -Path $OciPath) {
+                    Remove-Item -Recurse -Path $OciPath
+                    Write-Verbose "Cleared Oci local module cache" 
+                }
+                else {
+                    Write-Verbose "No Oci local module cache found" 
+                }
+            }            
+        }
+        
+        if ($TemplateSpecs) {            
+            $TSPath = Get-BicepNetCachePath -TemplateSpecs
+
+            if (($SubscriptionId) -and ($ResourceGroup) -and ($Spec) -and ($Version)) {
+                Remove-Item -Recurse -Path "$TSPath/$SubscriptionId/$ResourceGroup/$Spec/$Version"
+                Write-Verbose "Cleared version [$Version] of [$Spec] in [$ResourceGroup] in [$SubscriptionId] from local module cache"
+            }
+            elseif (($SubscriptionId) -and ($ResourceGroup) -and ($Spec)) {
+                Remove-Item -Recurse -Path "$TSPath/$SubscriptionId/$ResourceGroup/$Spec"
+                Write-Verbose "Cleared [$Spec] in [$ResourceGroup] in [$SubscriptionId] from local module cache"
+            }
+            elseif (($SubscriptionId) -and ($ResourceGroup)) {
+                Remove-Item -Recurse -Path "$TSPath/$SubscriptionId/$ResourceGroup"
+                Write-Verbose "Cleared [$ResourceGroup] in [$SubscriptionId] from local module cache"
+            }
+            elseif ($SubscriptionId) {
+                Remove-Item -Recurse -Path "$TSPath/$SubscriptionId"
+                Write-Verbose "Cleared [$SubscriptionId] from local module cache" 
+            }
+            else {
+                if (Test-Path -Path $TSPath) {
+                    Remove-Item -Recurse -Path $TSPath
+                    Write-Verbose "Cleared Template Spec local module cache" 
+                }
+                else {
+                    Write-Verbose "No Template Spec local module cache found" 
+                }
+            }            
+        }
+
+        if ($All) {            
```

Comment:
This can be replaced by adding an -and statement in the if's in line 49 and 76.
This may ease any refactoring as the removal of objects are in one place instead of two.

Reply:
Good suggestion! Went with -or though 👍